client/wp-admin-scripts/print-shipping-label-banner/shipping-banner/index.js

```diff
@@ -293,6 +297,26 @@ export class ShippingBanner extends Component {
 			const { orderId } = this.state;
 			const siteId = state.ui.selectedSiteId;
 
+			const wcsStoreUnsubscribe = wcsStore.subscribe( () => {
+				const latestState = wcsStore.getState();
+				const siteState =
+					latestState.extensions.woocommerce.woocommerceServices[
+						siteId
+					];
+				if ( siteState ) {
+					const orderState = siteState.shippingLabel[ orderId ];
+					if ( orderState ) {
+						if ( orderState.showPurchaseDialog ) {
+							if ( window.jQuery ) {
+								window
+									.jQuery( '#woocommerce-order-label' )
+									.show();
+							}
+							wcsStoreUnsubscribe();
```

Comment:
Since this is called inside `subscribe()` wouldn't this introduce infinite recursion? 

I took a look at https://redux.js.org/api/store/#subscribelistener point 2 and it says that the unsubscribe won't have any effect until the next dispatch call, so I think this is ok!

Reply:
Yep, breakpoint only got called once. 👍

Reply:
yeah, the docs mention that's the way to unsubscribe from a redux store :smiley: